cdap-ui/app/cdap/main.js

```diff
@@ -119,6 +131,52 @@ class CDAP extends Component {
     this.fetchSessionTokenAndUpdateState().then(this.setUIState);
   };
 
+  handlePageLevelError = (error) => {
+    // This function parses receiveing error messages and converts it to a
+    // format that page level error supports
+    let message = null;
+    let errorCode = null;
+    if (error.data) {
+      message = error.data;
+    } else if (typeof error.response === 'string') {
+      message = error.response;
+    }
+
+    if (error.statusCode) {
+      errorCode = error.statusCode;
+    } else {
+      // If we don't know about the error type, showing a 500 level error
+      errorCode = 500;
+    }
+    return { errorCode, message };
+  };
+
+  retrieveNamespace = () => {
+    this.setState({ retrievingNs: true });
+    this.namespaceSub = MyNamespaceApi.pollListTrue().subscribe(
```

Comment:
Is this even a function in namespace api?

Reply:
Also why do we need to poll for the list of namespaces?

Reply:
My bad, I was testing something and added another function with that name. I missed cleaning it up. Fixed.